packages/main/src/resolvers/conversation.test.ts

```diff
@@ -638,6 +638,116 @@ describe("when addressing conversations", () => {
     })
   })
 
+  it("an edited message will have an isRead field", async () => {
+    const orig = cache.getThreads(accountId)[0]
+    const message = testThread[1].attributes
+
+    expect(cache.getFlags(message.envelope.messageId).includes("//Seen")).toBe(
+      false
+    )
+
+    const part = message.struct![0] as imap.ImapMessagePart
+    const revisedContent = "What I meant to say was, hi."
+    const editMessage = composeEdit({
+      account,
+      content: {
+        type: "text",
+        subtype: "plain",
+        content: revisedContent
+      },
+      conversation: orig,
+      editedMessage: { envelope_messageId: message.envelope.messageId },
+      editedPart: {
+        content_id: part.id
+      },
+      resource: {
+        messageId: message.envelope.messageId,
+        contentId: part.id
+      }
+    })
+    editMessage.attributes.uid = 9000
+    const threadWithEdit = [...testThread, editMessage]
+    mock(Connection.prototype.fetch).mockImplementation(
+      mockFetchImplementation({ thread: threadWithEdit })
+    )
+    await sync(accountId, connectionManager)
+
+    const result = await request(
+      `
+        query getConversation($conversationId: ID!) {
+          conversation(id: $conversationId) {
+            presentableElements {
+              isRead
+              date
+              from {
+                name
+                mailbox
+                host
+              }
+              editedAt
+              editedBy {
+                name
+                mailbox
+                host
+              }
+              contents {
+                type
+                subtype
+                content
+              }
+            }
+          }
+        }
+      `,
+      { conversationId: testThread[0].attributes["x-gm-thrid"] }
+    )
+    expect(result).toMatchObject({
```

Comment:
This test could be made much shorter by requesting fewer fields, and only specifying the one field that you care about in the expect matcher.